src/devices/cpu/drcbex86.cpp

```diff
@@ -417,12 +421,30 @@ inline void drcbe_x86::emit_combine_z_flags(Assembler &a)
 {
 	// this assumes that the flags from the low 32-bit op are on the stack
 	// and the flags from the high 32-bit op are live
-	a.pushfd();                                                                         // pushf
-	a.mov(ecx, ptr(esp, 4));                                                            // mov   ecx,[esp+4]
-	a.or_(ecx, ~0x40);                                                                  // or    ecx,~0x40
-	a.and_(ptr(esp, 0), ecx);                                                           // and   [esp],ecx
-	a.popfd();                                                                          // popf
-	a.lea(esp, ptr(esp, 4));                                                            // lea   esp,[esp+4]
+	a.pushfd();
+
+	a.mov(ecx, dword_ptr(esp, 4)); // zero flag
+	a.or_(ecx, ~0x40);
+	a.and_(dword_ptr(esp, 0), ecx);
+
+	a.popfd();
+	a.lea(esp, ptr(esp, 4));
```

Comment:
Probably more complex to do given the code is already structured around using the stack, but this would get a performance benefit from avoiding `pushfd`/`popfd`, too.